src/kinetics/Reaction.cpp

```diff
@@ -253,11 +308,129 @@ std::string Reaction::equation() const
 void Reaction::setEquation(const std::string& equation, const Kinetics* kin)
 {
     parseReactionEquation(*this, equation, input, kin);
+
+    std::string rate_type = input.getString("type", "");
+    if (rate_type == "three-body") {
+        // state type when serializing
+        m_explicit_rate = true;
+    } else if (rate_type == "elementary") {
+        // user override
+        m_explicit_rate = true;
+        return;
+    } else if (rate_type == "two-temperature-plasma") {
+        // two-temperature-plasma reactions do not use third bodies
+        return;
```

Comment:
I'm confused by this comment -- this now seems to be permissible:
```py
>>> R = ct.Reaction(equation='OH + e + O2 => OH- + O2',
                    rate={'A': 1.0, 'b':0.0, 'Ea_gas': 3000, 'Ea_electron': 5000})
>>> print(repr(R))
OH + e + O2 => OH- + O2    <three-body-two-temperature-plasma>
```

Reply:
whoops. forgot to delete this one.